executable/build.go

```diff
@@ -37,9 +44,84 @@ type Build struct {
 func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
 	result := libcnb.NewBuildResult()
 
-	m, err := libjvm.NewManifest(context.Application.Path)
-	if err != nil {
-		return libcnb.BuildResult{}, fmt.Errorf("unable to read manifest in %s\n%w", context.Application.Path, err)
+	// Check if there is a top-level META-INF/MANIFEST.MF
+	manifest := filepath.Join(context.Application.Path, "META-INF", "MANIFEST.MF")
+	manifestExists := true
+	if _, err := os.Stat(manifest); errors.Is(err, os.ErrNotExist) {
+		manifestExists = false
+	}
+
+	var allJarFiles []string
+	m := properties.NewProperties()
+
+	if manifestExists {
+		var err error
+		m, err = libjvm.NewManifest(context.Application.Path)
+		if err != nil {
+			return libcnb.BuildResult{}, fmt.Errorf("unable to read manifest in %s\n%w", context.Application.Path, err)
+		}
+	} else {
+		// loop through the JAR files, collect them for the classpath and pick the first Main-Class
+		items, err := ioutil.ReadDir(context.Application.Path)
```

Comment:
I think that os.ReadDir() is preferred to ioutil in newer versions of Go.
Note this doesn't do recursive search so it would only list jars in current directory.
That's perfectly fine as the "Executable JAR" shouldn't be nested in some lib sub-directory.

Reply:
Thanks, I always wish vscode would just strike this through. And yes, I was intentionally not using a walker.